wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/DefaultWorkspaceValidator.java

```diff
@@ -123,6 +127,10 @@ private void validateEnv(Environment environment, String workspaceName) throws B
     private void validateMachine(MachineConfig machineCfg, String envName) throws BadRequestException {
         checkArgument(!isNullOrEmpty(machineCfg.getName()), "Environment %s contains machine with null or empty name", envName);
         checkNotNull(machineCfg.getSource(), "Environment " + envName + " contains machine without source");
+        if (apiEndpoint.startsWith("https://") && machineCfg.getSource().getLocation() != null) {
```

Comment:
Check location only if the type of the machine source is `recipe` [javadoc](https://github.com/eclipse/che/blob/ae4c5521bf2202740fc211ee3d2e94881e414d59/core/che-core-api-model/src/main/java/org/eclipse/che/api/core/model/machine/MachineSource.java#L19-L19)